cmd/snap-bootstrap/initramfs_mounts_state.go

```diff
@@ -80,3 +80,21 @@ func (mst *initramfsMountsState) UnverifiedBootModel() (*asserts.Model, error) {
 	}
 	return ma.(*asserts.Model), nil
 }
+
+// ModeenvFromModel generates a modeenv given the model and the snaps for the
+// current mode and recovery system of the initramfsMountsState.
+func (mst *initramfsMountsState) ModeenvFromModel(model *asserts.Model, snaps map[snap.Type]snap.PlaceInfo) (*boot.Modeenv, error) {
+	if mst.mode == "run" {
+		return nil, fmt.Errorf("internal error: initramfs should not write modeenv in run mode")
+	}
+	return &boot.Modeenv{
+		Mode:           mst.mode,
+		RecoverySystem: mst.recoverySystem,
+		Base:           snaps[snap.TypeBase].Filename(),
+		Model:          model.Model(),
+		BrandID:        model.BrandID(),
+		Grade:          string(model.Grade()),
+		// TODO:UC20: what about current kernel snaps, trusted boot assets and
```

Comment:
We may fill out the kernel snap, as we kind of guess what we booted with. However, we should not try to set the trusted boot assets nor the command lines and thus populating the modeenv with incorrect information that may be accidentally used somewhere.

Reply:
We agreed in the SU to not include the kernel snap here until we need it